src/request/actions/ActionCreator.ts

```diff
@@ -0,0 +1,32 @@
+import { Action } from 'redux';
+
+interface IActionCreatorAction<TPayload> extends Action {
+    payload: TPayload;
+}
+
+interface IActionCreatorBase<TPayload> {
+    type: string;
+    unwrap(action: Action): TPayload;
+}
+
+interface IActionCreator<TPayload> extends IActionCreatorBase<TPayload> {
+    (payload: TPayload): IActionCreatorAction<TPayload>;
+}
+
+export function createActionCreator<TPayload>(type: string): IActionCreator<TPayload> {
+    const actionCreator = (payload: TPayload) => {
+        return {
+            type: type,
+            payload: payload
```

Comment:
Also note that this `ActionCreator` is a bit of an experiment to try to elimination of duplicated strings (i.e. the action types) between smart component and reducer, while at the same time coupling action types to the data associated with them.  (The former would often be done through string constants, but I've yet to see anybody tackle the latter.)